subxt/src/backend/unstable/rpc_methods.rs

```diff
@@ -0,0 +1,890 @@
+    Invalid(ErrorEvent),
+}
+
+/// The runtime specification of the current block.
+///
+/// This event is generated for:
+///   - the first announced block by the follow subscription
+///   - blocks that suffered a change in runtime compared with their parents
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RuntimeVersionEvent {
+    /// Details about this runtime.
+    pub spec: RuntimeSpec,
+}
+
+/// This contains the runtime version information necessary to make transactions, and is obtained from
+/// the "initialized" event of `chainHead_follow` if the `withRuntime` flag is set.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RuntimeSpec {
+    /// Opaque string indicating the name of the chain.
+    pub spec_name: String,
+
+    /// Opaque string indicating the name of the implementation of the chain.
+    pub impl_name: String,
+
+    /// Opaque integer. The JSON-RPC client can assume that the Runtime API call to `Metadata_metadata`
+    /// will always produce the same output as long as the specVersion is the same.
+    pub spec_version: u32,
+
+    /// Opaque integer. Whenever the runtime code changes in a backwards-compatible way, the implVersion
+    /// is modified while the specVersion is left untouched.
+    pub impl_version: u32,
+
+    /// Opaque integer. Necessary when building the bytes of a transaction. Transactions that have been
+    /// generated with a different `transaction_version` are incompatible.
+    pub transaction_version: u32,
+
+    /// Object containing a list of "entry point APIs" supported by the runtime. Each key is an opaque string
+    /// indicating the API, and each value is an integer version number. Before making a runtime call (using
+    /// chainHead_call), you should make sure that this list contains the entry point API corresponding to the
+    /// call and with a known version number.
+    ///
+    /// **Note:** In Substrate, the keys in the apis field consists of the hexadecimal-encoded 8-bytes blake2
+    /// hash of the name of the API. For example, the `TaggedTransactionQueue` API is 0xd2bc9897eed08f15.
+    #[serde(with = "hashmap_as_tuple_list")]
+    pub apis: HashMap<String, u32>,
+}
+
+/// The operation could not be processed due to an error.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ErrorEvent {
+    /// Reason of the error.
+    pub error: String,
+}
+
+/// Indicate a new non-finalized block.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NewBlock<Hash> {
+    /// The hash of the new block.
+    pub block_hash: Hash,
+    /// The parent hash of the new block.
+    pub parent_block_hash: Hash,
+    /// The runtime version of the new block.
+    ///
+    /// # Note
+    ///
+    /// This is present only if the `with_runtime` flag is set for
```

Comment:
Do you mean with `with_runtime` the argument `runtime_updates: bool` that is passed to `chainhead_unstable_follow()`? I think we should use a uniform name adressing it throughout the code. 
Since it has an effect on the data in many struct, we could also think about making the `FollowEvent` generic over this flag (maybe with const generics). Just an idea. :)

Reply:
Yeah; we should prolly call it `with_runtime` elsewhere too because the spec calls it `withRuntime`!

I guess generics could be used to "guarantee" that we get a RuntimeEvent if we ask for it, and not if we don't, maybe? But the generics might complicate things more than it's worth because they would bleed out to anything else that cared about this stuff :)

Reply:
Yeah I agree that we should not complicate it more.